### DISCUSSION TITLE
Cold start postgresql

### DISCUSSION THREAD

**Discussion by friedrith** [vercel/community#3211] [Help] [answered]
### Description

I am using the new Postgres Database in beta from vercel.

I am using it like this, directly in a server component in next.js:

```tsx
const Event: React.FC<EventProps> = async ({ params }) => {
  const event = await db
    .selectFrom('events')
    .select(['id', 'title', 'slug', 'date', 'location'])
    .where('events.slug', '=', params.eventSlug)
    .executeTakeFirst()


   //...

}
```

I understood there are some [cold starts](https://vercel.com/docs/storage/vercel-postgres/limits#vercel-postgres-cold-starts) but it is triggering some errors on my application. 

<img width="390" alt="image" src="https://github.com/vercel/vercel/assets/4005226/91b1634d-577f-48f7-8b0c-c127e3f16ef5">

If I refresh the page, it works again for 5-10 minutes. Then again the first person to go on the page will experience an error 500.

In the logs, I found an error that pushes me to think it is coming from postgres:

```
"servername":"ep-fragrant-dream-984147-pooler.us-east-1.postgres.vercel-storage.com"
```

Do you know how I could avoid this situation (using some caches maybe)?

**Comment by swarnava** [comment, marked as answer]
Hi,

Thank you for raising this issue.
It appears that the error is indeed related to cold starts of the underlying Neon database. A fix should now be live for the “Couldn’t connect to compute node”. Let us know if you are still facing issue.

**Reply by friedrith** [reply]
Awesome! So quick! The problem seems resolved. I will try several times today to be sure.
